model/import/SampleTemplateDownload.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+/**
+ * DownloadSampleCsv Controller provide actions to download sample CSV
+ *
+ * @author Chinnu Francis - TAO Team - {@link http://www.tao.lu}
+ * @package taoQtiItem
+ * @license GPLv2  http://www.opensource.org/licenses/gpl-2.0.php
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoQtiItem\model\import;
+
+use oat\taoQtiItem\model\import\Repository\CsvTemplateRepository;
+use oat\taoQtiItem\model\import\Repository\MetadataRepository;
+use oat\taoQtiItem\model\import\Repository\TemplateRepositoryInterface;
+use oat\taoQtiItem\model\import\Parser\TemplateHeaderParser;
+use oat\taoQtiItem\model\import\Factory\CsvTemplateSampleLineFactory;
+use oat\oatbox\service\ConfigurableService;
+use Zend\ServiceManager\ServiceLocatorAwareInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Message\ResponseInterface;
+use function GuzzleHttp\Psr7\stream_for;
+use common_exception_Error;
+use common_exception_MissingParameter;
+
+class SampleTemplateDownload extends ConfigurableService implements ServiceLocatorAwareInterface
+{
+    /**
+     * @throws common_exception_Error
+     */
+    public function download(ServerRequestInterface $request, ResponseInterface $response)
+    {
+        $queryParams = $request->getQueryParams();
+        if (!$queryParams['uri']) {
+            throw new common_exception_MissingParameter('uri', __METHOD__);
+        }
+        $uri = $queryParams['uri'];
+        $metaDataArray = $this->getMetadataRepository()->findMetadataByClassUri($uri);
+    
+        $template = $this->getTemplateRepository()->findById(CsvTemplateRepository::DEFAULT);
+        $headers = $this->getTemplateHeaderParser()->parse($template, $metaDataArray);
+
+        $templateSampleLines = $this->getTemplateSampleLines()->getSampleLines($template);
+
+        $className = $this->getMetadataRepository()->getClassName($uri);
+        $filename = $this->getFileName($className);
+
+        $csvContent = $this->getCsvContent($headers, $templateSampleLines);
+        return $response
+            ->withHeader('Content-Type', 'text/csv')
+            ->withHeader("Content-Disposition", "attachment; filename=" . $filename)
+            ->withBody(stream_for($csvContent));
+    }
+
+    private function getCsvContent($headers, $templateSampleLines): string
+    {
+        $file = fopen('php://temp', 'rw');
+        fputcsv($file, $headers);
+        foreach ($templateSampleLines as $row) {
+            fputcsv($file, $row);
+        }
+        rewind($file);
+        $csvContent = stream_get_contents($file);
```

Comment:
Why do you convert file(stream) into string and than later doing opposite operation https://github.com/oat-sa/extension-tao-itemqti/pull/1751/commits/a5d20579a5fffe53f86f4edde62ee7f725f09502#diff-f600f96da6a7af45888c3bab160423fc3fdac0b5d30a26db7e460dcdf2c31993R54 ?

Reply:
done